src/test/regress/expected/function_propagation.out

```diff
@@ -316,6 +316,387 @@ BEGIN
     return 1;
 END;
 $$;
+-- Show that functions are propagated (or not) as a dependency
+-- Function as a default column
+BEGIN;
+CREATE OR REPLACE FUNCTION func_in_transaction_def()
+RETURNS int
+LANGUAGE plpgsql AS
+$$
+BEGIN
+    return 1;
+END;
+$$;
+-- Function shouldn't be propagated within transaction
+SELECT pg_identify_object_as_address(classid, objid, objsubid) from citus.pg_dist_object where objid = 'function_propagation_schema.func_in_transaction_def'::regproc::oid;
+ pg_identify_object_as_address
+---------------------------------------------------------------------
+(0 rows)
+
+CREATE TABLE table_to_prop_func(id int, col_1 int default func_in_transaction_def());
+SELECT create_distributed_table('table_to_prop_func','id');
+ create_distributed_table
+---------------------------------------------------------------------
+
+(1 row)
+
+-- Function should be propagated after distributing the table that depends on it
+SELECT pg_identify_object_as_address(classid, objid, objsubid) from citus.pg_dist_object where objid = 'function_propagation_schema.func_in_transaction_def'::regproc::oid;
+                     pg_identify_object_as_address
+---------------------------------------------------------------------
+ (function,"{function_propagation_schema,func_in_transaction_def}",{})
+(1 row)
+
+SELECT * FROM run_command_on_workers($$SELECT pg_identify_object_as_address(classid, objid, objsubid) from citus.pg_dist_object where objid = 'function_propagation_schema.func_in_transaction_def'::regproc::oid;$$) ORDER BY 1,2;
```

Comment:
does this really show that the function is propagated remotely? It seems the result is NULL?
 I guess we should run `run_command_on_workers` after we commit the tx?

Reply:
I've updated them to check after committing transaction.